dask/dataframe/io.py

```diff
@@ -724,10 +736,59 @@ def one_path_one_key(path, key, start, stop, columns, chunksize, lock):
     if (start != 0 or stop is not None) and len(keys) > 1:
         raise NotImplementedError(read_hdf_error_msg)
     from .multi import concat
-    return concat([one_path_one_key(path, k, start, s, columns, chunksize, lock)
+    return concat([one_path_one_key(path, k, start, s, columns, chunksize,
+                                    lock, division_column)
                    for k, s in zip(keys, stops)])
 
 
+def _find_divisions(division_column, start, stop, chunksize, path, key, lock):
+    divisions = list()
+    starts = [start]
+    stops = list()
+    row = start
+    if isinstance(division_column, string_types):
+        _div_col = lambda x: x[division_column]
+    elif division_column:
+        _div_col = lambda x: pd.Series(x.index)
+    stop = _get_last_row(path, key, lock, stop)
+    data = _pd_read_hdf(path, key, lock,
+                        {'start': row, 'stop': row + 1})
+    divisions.append(_div_col(data).iloc[0])
+    for row in range(start + chunksize, stop, chunksize):
+        data = _pd_read_hdf(path, key, lock,
+                            {'start': row - 1, 'stop': row + 1})
+        while _div_col(data).iloc[0] == _div_col(data).iloc[1]:
+            row += 1
+            if row >= stop:
+                break 
+            data = _pd_read_hdf(path, key, lock,
+                                {'start': row - 1, 'stop': row + 1})
```

Comment:
I suspect we can make this much faster with the following:

1.  Use `pd.read_hdf` instead of the internal `_pd_read_hdf` which does a lot of extra stuff we don't need here, notably locking
2.  Read lots of rows at once.  I suspect that the cost difference between reading one row and one hundred rows is trivial, and it would save us many iterations
3.  Read just the column that we care about rather than all of the columns (some datasets have hundreds of columns that we really don't want to have to deal with)